src/EncoderDecoder/Huffman/HuffmanEncoderDecoder.cpp

```diff
@@ -0,0 +1,252 @@
+#include "EncoderDecoder/Huffman/HuffmanEncoderDecoder.h"
+
+HuffmanEncoderDecoder::NodeVector HuffmanEncoderDecoder::getFrequencies(
+	const std::string &plaintext) {
+
+	// Initialize frequencies
+	NodeVector frequencies;
+	frequencies.reserve(256);
+	for(int c = 0; c < 256; c++) {
+		frequencies.push_back(std::make_unique<Node>(Node(c)));
+	}
+
+	// Count frequency of each char in plaintext
+	std::for_each(plaintext.cbegin(), plaintext.cend(), [&](unsigned char c) {
+		auto &fp = frequencies[c];
+		(*fp)++;
+	});
+
+	return frequencies;
+}
+
+HuffmanEncoderDecoder::NodePtr HuffmanEncoderDecoder::buildPrefixlessTree(
+	NodeVector &frequencies) {
+
+	// Sort by ascending order of frequency
+	std::sort(frequencies.begin(),
+		frequencies.end(),
+		[](const NodePtr &n1, const NodePtr &n2) {
+			return n1->frequency < n2->frequency;
+		});
+
+	// Find the first element with non-zero frequency
+	auto activeSymbols = std::upper_bound(
+		frequencies.begin(), frequencies.end(), 0, [](int value, NodePtr &n) {
+			return n->frequency > value;
+		});
+
+	// Copy to list non-zero frequency elements
+	std::list<NodePtr> frequenciesList;
+	std::copy(std::make_move_iterator(activeSymbols),
```

Comment:
Might be because I’m not familiar enough with all algorithms yet but why `copy(move_iterator` instead of the `move()` algorithm?

Reply:
I didn't know `std::move` had an overload similar to other STL algorithms, but from what I've seen [here](https://stackoverflow.com/a/26433162/11932596), it does.

TODO